jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,659 @@
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.exclude(new InetAddressPredicate(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern} in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void excludeInetAddress(String... patterns)
+    {
+        for (String p : patterns)
+            exclude(InetAddressPattern.from(p));
+    }
+
+    /**
+     * {@link IncludeExclude#include(Object) Include} a {@link ConnectorAddrMethodPathPredicate}.
+     * If only a method or pathSpec is passed, then this method optimizes by converting this
+     * call to {@link #includeMethod(String...)} or {@link #include(PathSpec...)}.
+     *
+     * @param connectorName optional name of a connector or {@code null}.
+     * @param addressPattern optional InetAddress pattern to test against the remote address or {@code null}.
+     * @param method optional method or {@code null}.
+     * @param pathSpec optional pathSpec to test against the {@link Request#getPathInContext(Request) pathInContext} or {@code null}.
+     */
+    public void include(String connectorName, String addressPattern, String method, PathSpec pathSpec)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+
+        if (connectorName == null && addressPattern == null)
+        {
+            if (StringUtil.isNotBlank(method) && pathSpec == null)
+                includeMethod(method);
+            else if (StringUtil.isBlank(method) && pathSpec != null)
+                include(pathSpec);
+            include(new ConnectorAddrMethodPathPredicate(connectorName, InetAddressPattern.from(addressPattern), method, pathSpec));
+        }
+        else
+        {
+            include(new ConnectorAddrMethodPathPredicate(connectorName, InetAddressPattern.from(addressPattern), method, pathSpec));
+        }
+    }
+
+    /**
+     * {@link IncludeExclude#exclude(Object) Exclude} a {@link ConnectorAddrMethodPathPredicate}.
+     * If only a method or pathSpec is passed, then this method optimizes by converting this
+     * call to {@link #excludeMethod(String...)} or {@link #exclude(PathSpec...)}.
+     *
+     * @param connectorName optional name of a connector or {@code null}.
+     * @param addressPattern optional InetAddress pattern to test against the remote address or {@code null}.
+     * @param method optional method or {@code null}.
+     * @param pathSpec optional pathSpec to test against the {@link Request#getPathInContext(Request) pathInContext} or {@code null}.
+     */
+    public void exclude(String connectorName, String addressPattern, String method, PathSpec pathSpec)
```

Comment:
Ditto, this shouldn't be here.  Too specific for a small handful of cases.

Reply:
ditto it is used and there is a reason for it, as per the javadoc.